core/templates/pages/contributor-dashboard-page/modal-templates/question-suggestion-editor-modal.component.ts

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview component for question suggestion editor modal.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { NgbActiveModal, NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { AlertsService } from 'services/alerts.service';
+import { AppConstants } from 'app.constants';
+import { MisconceptionSkillMap } from 'domain/skill/MisconceptionObjectFactory';
+import { Question } from 'domain/question/QuestionObjectFactory';
+import { QuestionUndoRedoService } from 'domain/editor/undo_redo/question-undo-redo.service';
+import { Skill } from 'domain/skill/SkillObjectFactory';
+import { SkillDifficulty } from 'domain/skill/skill-difficulty.model';
+import { State } from 'domain/state/StateObjectFactory';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { ConfirmQuestionExitModalComponent } from 'components/question-directives/modal-templates/confirm-question-exit-modal.component';
+import { QuestionsOpportunitiesSelectDifficultyModalComponent } from 'pages/topic-editor-page/modal-templates/questions-opportunities-select-difficulty-modal.component';
+import { ContextService } from 'services/context.service';
+import { ContributionAndReviewService } from '../services/contribution-and-review.service';
+import { ImageLocalStorageService } from 'services/image-local-storage.service';
+import { QuestionSuggestionBackendApiService } from 'pages/contributor-dashboard-page/services/question-suggestion-backend-api.service';
+import { QuestionValidationService } from 'services/question-validation.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+
+@Component({
+  selector: 'oppia-question-suggestion-editor-modal',
+  templateUrl: './question-suggestion-editor-modal.component.html'
+})
+export class QuestionSuggestionEditorModalComponent
+  extends ConfirmOrCancelModal implements OnInit {
+  @Input() question: Question;
+  @Input() questionStateData: State;
+  @Input() questionId: string;
+  @Input() skill: Skill;
+  @Input() skillDifficulty: SkillDifficulty | string;
```

Comment:
Why can this be both `SkillDifficulty` and `string`?

Reply:
we are taking skillDifficulty as input  so it can we of both type check function `done()` if  isEditing is true then skillDifficulty is used as string otherwise in else part it is used as SkillDifficulty.

Reply:
fixed in pr #15859, PTAL